runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -341,6 +342,36 @@ AOTCacheAOTHeaderRecord::create(uintptr_t id, const TR_AOTHeader *header)
    return new (ptr) AOTCacheAOTHeaderRecord(id, header);
    }
 
+ThunkSerializationRecord::ThunkSerializationRecord(uintptr_t id, const void *signature, size_t signatureSize,
+                                                   const void *thunkCode,  size_t thunkCodeSize) :
+   AOTSerializationRecord(size(signatureSize, thunkCodeSize), id, AOTSerializationRecordType::Thunk),
+   _signatureSize(signatureSize),
+   _thunkCodeSize(thunkCodeSize)
+   {
+   memcpy((void *)this->signature(), signature, signatureSize);
```

Comment:
In this case it's a `const void *` to `void *` cast, so it's necessary. It's done in the other serialization records with variable-sized `_data` fields as well.

Reply:
Oh, never mind. Some of the others use the fields directly.

Reply:
I see. It's done with this style of cast in, for instance, the `AOTCacheListRecord` constructor because the final `_varSizedData` field has to hold multiple distinct variable-sized fields in it, so it's easier to cast the accessors inline as I do it here than to calculate the proper offsets in the `_varSizedData` field again.